helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java

```diff
@@ -573,35 +618,36 @@ public void resetPartition(String clusterName, String instanceName, String resou
     // check the instance is alive
     LiveInstance liveInstance = accessor.getProperty(keyBuilder.liveInstance(instanceName));
     if (liveInstance == null) {
-      throw new HelixException(
-          "Can't reset state for " + resourceName + "/" + partitionNames + " on " + instanceName
-              + ", because " + instanceName + " is not alive");
+      // check if the instance has never exist in the cluster
+      String instancePath = PropertyPathBuilder.instance(clusterName, instanceName);
+      if (!_zkClient.exists(instancePath)) {
+        throw new HelixException(
+            resetPartitionLogHelper(clusterName, instanceName, resourceName, partitionNames, null,
+                ResetPartitionFailureReason.INSTANCE_NOTALIVE));
+      } else {
+        throw new HelixException(
+            resetPartitionLogHelper(clusterName, instanceName, resourceName, partitionNames, null,
+                ResetPartitionFailureReason.INSTANCE_NONEXISTENT));
```

Comment:
Nit, just about style :) To reduce the duplicate code, I would do something like this
```
String failureReason = _zkClient.exists(instancePath) ? ResetPartitionFailureReason.INSTANCE_NONEXISTENT) :  ResetPartitionFailureReason.INSTANCE_NOTALIVE);
throw new HelixException(...);
```